lib/geordi/gitlinear.rb

```diff
@@ -0,0 +1,201 @@
+        No staged changes. Will create an empty commit.
+      WARNING
+
+      issue = choose_issue
+      if issue
+        create_commit "[#{issue['identifier']}] #{issue['title']}", "Issue: #{issue['url']}", *git_args
+      end
+    end
+
+    def run_branch(from_master: false)
+      issue = choose_issue || Interaction.fail('No issue selected.')
+
+      branch_list_string = if Util.testing?
+                             ENV['GEORDI_TESTING_GIT_BRANCHES'] || ''
+                           else
+                             `git branch --format="%(refname:short)"`
+                           end
+
+      if branch_list_string.nil? || branch_list_string.strip.empty?
+        Interaction.fail 'Could not determine local git branches.'
+      end
+
+      new_branch_name = "#{issue['branchName']}"
+
+      local_branches = branch_list_string.split("\n")
+
+      branch_name = local_branches.find { |branch_name| branch_name == new_branch_name }
+      branch_name ||= local_branches.find { |branch_name| branch_name.include? issue['identifier'].to_s }
+
+      if branch_name.present?
+        checkout_branch branch_name, new_branch: false
+      else
+        checkout_branch new_branch_name, new_branch: true, from_master: from_master
+      end
+    end
+
+    private
+
+    attr_accessor :highline, :settings
+
+    def choose_issue
+      loading_message = 'Connecting to Linear ...'
+      print(loading_message)
+      issues = applicable_issues
+      reset_loading_message = "\r#{' ' * (loading_message.length + issues.length)}\r"
+
+      Geordi::Interaction.fail('No issues to offer.') if issues.empty?
+
+      if Util.testing?
+        return issues[0]
+      end
+
+      highline.choose do |menu|
+        menu.header = 'Choose an issue'
+
+        issues.each do |issue|
+          state = issue['state']['name']
+          if issue['assignee']
+            assignee = issue['assignee']['name']
+            assignee_is_me = issue['assignee']['isMe']
+          else
+            assignee = "none"
+            assignee_is_me = false
+          end
+
+          if state == 'In Progress'
+            state = HighLine::GREEN + state + HighLine::RESET
+          else
+            state = HighLine::RED + state + HighLine::RESET
+          end
+
+          state += HighLine::BOLD if assignee_is_me
+
+          label = "(#{assignee}, #{state}) #{issue['title']}"
+          label = bold(label) if assignee_is_me
+
+          menu.choice(label) { return issue }
+        end
+
+        menu.hidden ''
+        print reset_loading_message # Once menu is build
+      end
+
+      nil # Return nothing
+    end
+
+    def create_commit(title, description, *git_args)
+      extra = highline.ask("\nAdd an optional message").strip
+      title << ' - ' << extra if extra != ''
+      Util.run!(['git', 'commit', '--allow-empty', '-m', title, '-m', description, *git_args])
+    end
+
+    def applicable_issues
```

Comment:
Please name this method so it is clear what it does:
```suggestion
    def fetch_linear_issues
```